### ISSUE TITLE
autoLabel and labelFormat working strange

### ISSUE THREAD

**Issue by andrewkazan** [emotion-js/emotion#3080]
**I'm using CRA with type script 4.7.4 and dependencies:**

```
"dependencies": {
  "@emotion/is-prop-valid": "^0.8.1",
  "@emotion/react": "^11.9.3",
  "@emotion/styled": "^11.6.0",
  "react": "^18.2.0",
```

```
"devDependencies": {
  "@babel/plugin-proposal-class-properties": "^7.18.6",
  "@babel/plugin-proposal-decorators": "^7.18.10",
  "@babel/plugin-transform-flow-strip-types": "^7.18.9",
  "@babel/preset-env": "7.18.10",
  "@babel/preset-react": "7.18.6",
  "@babel/preset-typescript": "7.18.6",
  "@emotion/babel-plugin": "^11.10.5",
```

**My .babelrc:**

```
{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-typescript",
    [
      "@babel/preset-react",
      {
        "runtime": "automatic"
      }
    ]
  ],
  "plugins": [
    [
      "@emotion",
      {
        "autoLabel": "always",
        "labelFormat": "[local]"
      }
    ],
    "babel-plugin-macros",
    "@babel/plugin-transform-flow-strip-types",
    [
      "@babel/plugin-proposal-decorators",
      {
        "legacy": true
      }
    ],
    [
      "@babel/plugin-proposal-class-properties",
      {
        "loose": true
      }
    ]
  ]
}
```

When I using dev mode without settings in .babelrc, I see class names like this:

`class="css-4w7amp-Container evgra233"`

but some class names have lot same names

`class="css-1ow6fxa-SensorWrapper-SensorWrapper-SensorWrapper eqhyoxx16"`

If I made settings in .babelrc for example "labelFormat": "my-class-name-[local]", I got change class names only which contain lot of same names

`class="css-8omzlq-SensorWrapper-SensorWrapper-my-class-name-SensorWrapper eqhyoxx16"`

but other names of classes do not change, "Container" is still "Container"

`class="css-4w7amp-Container evgra233"`

From build I got names with my prefix only from names where is contain lot of same names

`class="css-uafe6e-SensorWrapper eqhyoxx16"`

but other names have dynamic names without my prefix

`class="css-andsd2 eqhyoxx17"`

**Behavior is strange and I have questions:**

1. Why some class names do not change?
2. Why some class names have lot of same names?
3. Why in build with flag in .babelrc "autoLabel": "always" I got labels only which was with lot of names?

_screenshot:_ 
**1. dev mode - "autoLabel": "always",  "labelFormat": "my-class-name-[local]"** 
<img width="465" alt="From dev flag always" src="https://github.com/emotion-js/emotion/assets/31125066/b3653d46-846d-4a34-8b00-9d28b2df26a7">

**2. dev mode - "autoLabel": "never" but it contain variable prefix**
<img width="456" alt="From dev flag never" src="https://github.com/emotion-js/emotion/assets/31125066/59af9b23-5cdc-470b-bd07-35cdbbd937fa">

**3. build - "autoLabel": "always", but some has variable prefix, some not**
<img width="452" alt="From build flag always" src="https://github.com/emotion-js/emotion/assets/31125066/0678d813-72a0-4b56-a5a7-01932effca2d">

**Comment by JarodFlorence** [comment]
@andrewkazan 

2. Why some class names have lots of the same names?
This kind of issue typically occurs when you are reusing a styled component inside itself.  Using styled components in a recursive manner causes this issue. For example, 

```
const SensorWrapper = styled.div`
  /* styles */
`;

const Container = styled(SensorWrapper)`
  /* additional styles */
`;
```

If you have the same problem in your code, the generated class name will contain repeated labels.

**Comment by andrewkazan** [comment]
Thank you for answer!

About question 2. No I'm using like that:

```
export const SensorWrapper = styled.div<I.SensorI & I.SensorDraggableI>`
  width: 200px;
  height: 112px;
  border-radius: 4px;
  padding: 12px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 6px;
```

I found reason where not working "autoLabel". It's import.
Before I did importing and it's not working:

`import styled from '@emotion/styled/macro'; `

Now, I changed import and it's woking good: 

import styled from '@emotion/styled';

"autoLabel" and "labelFormat" are working.

But at some items stay lot of class names. But It is not problem:

<img width="493" alt="Screenshot 2023-07-21 at 12 39 45" src="https://github.com/emotion-js/emotion/assets/31125066/0d474940-692c-4a69-90b0-0ba4220924b6">
